app/controllers/application_controller.rb

```diff
@@ -1,6 +1,41 @@
 class ApplicationController < ActionController::API
 
   private
+  set_current_tenant_through_filter
+  before_action :set_the_current_tenant
+
+  def set_the_current_tenant
+    return unless ENV["ENFORCE_TENANCY"]
+    return unless ident = user_identity
+
+    tenant = Tenant.find_by(:external_tenant => ident)
+    if tenant
+      set_current_tenant(tenant)
+    else
+      render :json => { :errors => "Unauthorized" }, :status => :unauthorized
+    end
+  end
+
+  def user_account_number
+    Base64.decode64(request.headers.fetch_path("x-rh-identity", "identity", "account_number"))
+  end
+
+  def user_identity
```

Comment:
Should this be cached?

Reply:
We're going to move this to a common gem so we can take care of that there